Lib/test/test_wave.py

```diff
@@ -196,6 +197,22 @@ def test_read_wrong_sample_width(self):
         with self.assertRaisesRegex(wave.Error, 'bad sample width'):
             wave.open(io.BytesIO(b))
 
+    @skip_unless_working_chmod
+    def test_write_to_protected_file(self):
+        # gh-136523: Wave_write.__del__ should not throw
+        stderr = io.StringIO()
+        sys.stderr = stderr
+        try:
+            try:
+                with unwritable_filepath() as path:
+                    with wave.open(path, "wb"):
+                        pass
+            except PermissionError:
+                pass
+            self.assertEqual(stderr.getvalue(), "")
```

Comment:
Call `test.support.gc_collect()` before checking the output. On other Python implementations (like PyPy) an object may not be deleted immediately after it becomes unused.